wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ViewModel/ConversationListViewControllerViewModel+ZMConversationListObserver.swift

```diff
@@ -43,11 +42,8 @@ extension ConversationListViewController.ViewModel {
         }
     }
 
-    func updateArchiveButtonVisibility() {
-        viewController?.updateArchiveButtonVisibilityIfNeeded(showArchived: ZMConversationList.hasArchivedConversations)
-    }
-
     var hasArchivedConversations: Bool {
-        return conversationListType.hasArchivedConversations
+        guard let contextProvider = userSession as? ContextProvider else { return false }
+        return ZMConversationList.archivedConversations(inUserSession: contextProvider).count > 0
```

Comment:
In general a `count > 0` indicates, that only a `.isEmpty` can be used. The benefit is, that this has a complexity of _O(1)_ instead of _O(n)_.

Reply:
The return type of `archivedConversations` is `ZMConversationList`, which is a subclass of `NSArray` 🤔.
Still, there is no `isEmpty` function.

Reply:
Oh, sorry I've missed this point. Probably one could check [`firstObject == nil`](https://developer.apple.com/documentation/foundation/nsarray/1412852-firstobject), to get the same complexity, but for me `count > 0` is totally fine then.